src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -180,13 +180,18 @@ static ClassDesc ofDescriptor(String descriptor) {
      * @jvms 4.4.1 The CONSTANT_Class_info Structure
      */
     default ClassDesc arrayType() {
-        int depth = arrayDepth(descriptorString());
+        String desc = descriptorString();
+        int depth = arrayDepth(desc);
         if (depth >= MAX_ARRAY_TYPE_DESC_DIMENSIONS) {
             throw new IllegalStateException(
                     "Cannot create an array type descriptor with more than " +
                     MAX_ARRAY_TYPE_DESC_DIMENSIONS + " dimensions");
         }
-        return arrayType(1);
+        String newDesc = new StringBuilder(desc.length() + 1).append('[').append(desc).toString();
```

Comment:
This could use [`String​::concat​(String)`], which uses the same underlying code path as the `invokedynamic`‑based `String` concatenation:
```suggestion
        String newDesc = "[".concat(desc);
```

[`String​::concat​(String)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/String.html#concat(java.lang.String)